src/diagnostic.rs

```diff
@@ -14,6 +14,18 @@ use deno_ast::ModuleSpecifier;
 use deno_ast::SourceRange;
 use deno_ast::SourceTextInfo;
 
+#[derive(Debug, Clone)]
+pub struct LintQuickFixChange {
+  pub new_text: Cow<'static, str>,
+  pub range: SourceRange,
+}
+
+#[derive(Debug, Clone)]
+pub struct LintQuickFix {
+  pub description: Cow<'static, str>,
+  pub changes: Vec<LintQuickFixChange>,
+}
```

Comment:
Correct me if I'm wrong, but my understanding is that the user of `deno_lint` crate will use these quickfix information as suggestions, and end users will decide what quickfix to apply probably via LSP action, or some other interactive way - is this correct?

What I'm imagining is that, people who work with very large code base may want to apply quickfixes automatically without any human intervention. In this scenario, they probably would like something like `deno lint --rule no-window-prefix` --autofix, but one problem with this would be that one fix needs to be picked in case there are multiple candidates for quickfixes, which would require human intervention.
I think one possible solution for this is give each quick fix a unique ID, and let end users specify a quickfix ID they want to use e.g. `deno lint --rule no-window-prefix --autofix use-globalthis`
So in any case, what do you think about giving quickfix ID to each quickfix definition?

Reply:
Yeah in the LSP they'll potentially get multiple options.

For `--fix`, what I'm doing at the moment (in my local branch of cli, didn't push yet) is only the first quick fix gets applied if there are multiple options. So for example, In the window -> globalThis or window -> self case it only will apply the window -> globalThis change.

I think adding ids would be a nice solution in the future, but that we should hold off on it until there's demand for it. I think the current PR is a good starting point to build off of in the future.

In the meantime I'll update the docs in this crate to explain this scenario.

Reply:
OK, that makes sense. Let's revisit it when the need for it arises.